### ISSUE TITLE
Add the option of copying jars instead of using .ivy repo

### ISSUE THREAD

**Issue by sclasen** [sbt/sbt-start-script#11]
This will help keep sizes down on Heroku, etc.  Play 2 has a stage task that does the copying so you can grab the code from there...

 val playStage = TaskKey[Unit]("stage")
  val playStageTask = (baseDirectory, playPackageEverything, dependencyClasspath in Runtime, target, streams) map { (root, packaged, dependencies, target, s) =>

```
import sbt.NameFilter._

val staged = target / "staged"

IO.delete(staged)
IO.createDirectory(staged)

val libs = dependencies.filter(_.data.ext == "jar").map(_.data) ++ packaged

libs.foreach { jar =>
  IO.copyFile(jar, new File(staged, jar.getName))
}

val start = target / "start"
IO.write(start,
  """|#! /usr/bin/env sh
     |
     |java "$@" -cp "`dirname $0`/staged/*" play.core.server.NettyServer `dirname $0`/..
     |""".stripMargin)

"chmod a+x %s".format(start.getAbsolutePath) !

s.log.info("")
s.log.info("Your application is ready to be run in place: target/start")
s.log.info("")

()
```

  }

**Comment by havocp** [comment]
This will probably turn out to be a dumb question, but why does this save space? Doesn't it make another copy and thus use more space? Or would you modify things to avoid including .ivy2 in the slug and then we're saving the extra .xml and other non-jar files? How will Heroku know whether to include .ivy2 in the slug?

**Comment by sclasen** [comment]
Yeah so making it an optional thing would be good so folks could choose to not use .ivy2...since buildpacks are now open source, and user visible it is simple to fork the existing scala buildpack and drop .ivy2 from the slug to reduce size.

**Comment by imikushin** [comment]
+1 play stage task is nice: you just grab `target/start` and `staged` dir, archive it and you are good to go!

**Comment by nelsonjchen** [comment]
This might be fixed for Heroku by now. I think I saw something about ivy's cache being dropped in my deploy. Is this feature still in request?

**Comment by havocp** [comment]
So I guess the feature here would be that if some setting is present (like stageJars := true or something) we copy jars into staged/ and then make that the classpath? Maybe at that point Play could just use this plugin?

It sounds fine to me, I haven't had a chance to code it up though.

**Comment by stig** [comment]
This is not fixed in heroku yet, to my knowledge, but it is arguably where it _should_ be fixed.

**Comment by havocp** [comment]
fwiw all Heroku's knowledge of sbt is in https://github.com/heroku/heroku-buildpack-scala if anyone wants to hack on that. You can also use a custom version for your own app.

**Comment by stig** [comment]
We already have a fork of the scala build pack to get around the problem that every so often the .ivy2 cache gets too big and slug deployment fails. But our hack just blows away the .ivy2 cache at the start of the build and as such makes the build slower to the extent that it sometimes times out... So we haven't submitted a pull request. Here's our fork, if people are interested anyway: https://github.com/net-a-porter-mobile/heroku-buildpack-scala
